src/main/java/com/javastudy/coworkings/dao/jdbc/JdbcCoworkingDao.java

```diff
@@ -46,22 +50,51 @@ public Coworking getById(long id) {
             try (ResultSet resultSet = preparedStatement.executeQuery()) {
 
                 if (!resultSet.next()) {
-                    throw new RuntimeException("Coworking with id: " + id + " isn't found");
+                    throw new RuntimeException("Co-working with id: " + id + " isn't found");
                 }
                 Coworking coworking = COWORKING_ROW_MAPPER.rowMap(resultSet);
 
                 if (resultSet.next()) {
-                    throw new RuntimeException("Coworking's with id: " + id + " are several");
+                    throw new RuntimeException("Co-working's with id: " + id + " are several");
                 }
 
                 return coworking;
             }
         } catch (SQLException e) {
             logger.error("SQL Failed: {}", GET_COWORKING_BY_ID);
-            throw new RuntimeException("Coworking with id: " + id + "wasn't found", e);
+            throw new RuntimeException("Co-working with id: " + id + "wasn't found", e);
+        }
+    }
+
+    @Override
+    public List<Coworking> getByCity(String city){
+        try (Connection connection = dataSource.getConnection();
+             PreparedStatement preparedStatement = connection.prepareStatement(GET_COWORKING_BY_CITY)) {
+            List<Coworking> coworkings = new ArrayList<>();
+            preparedStatement.setString(1, city);
+
+            try (ResultSet resultSet = preparedStatement.executeQuery()) {
+
+                while (resultSet.next()) {
+                    Coworking coworking = COWORKING_ROW_MAPPER.rowMap(resultSet);
+                    coworkings.add(coworking);
+                }
+
+                if (coworkings.size() == 0) {
+                    logger.warn("No co-workings are found for the city \"{}\"", city);
+                } else {
+                    logger.info("{} coworkings were found for the city \"{}\"", coworkings.size(), city);
+                }
+
+                return coworkings;
+            }
+        } catch (SQLException e) {
+            logger.error("SQL Failed: {}", GET_COWORKING_BY_CITY);
+            throw new RuntimeException("Connection to database is not available . It is not possible to search co-working by city: " + city, e);
```

Comment:
Connection to database is not available 

How do you know? Maybe SQL is wrong? Or you try to getDouble from ResultSet where the column is a string?